File: typescript/space-age/space-age.ts
const earthYears = {
  mercury: 0.2408467,
  venus: 0.61519726,
  earth:  1.0,
  mars:  1.8808158,
  jupiter:  11.862615,
  saturn:  29.447498,
  uranus:  84.016846,
  neptune:  164.79132,
} as const

type Planet = keyof typeof earthYears

function floatToFixed(float: number, toFix: number): number {
  return Number.parseFloat(float.toFixed(toFix))
}

function secondsToYears(seconds: number): number {
  return seconds / (60 * 60 * 24 * 365.245)
}

export function age(planet: Planet, seconds: number): number {
  const factor = earthYears[planet]

  if (!factor) {
    throw new Error('Planet not found')
  }

  const ageInYears = secondsToYears(seconds)
  const result = ageInYears / factor
  console.log('result', result)

  return floatToFixed(result, 2)
}

File: go/space-age/space_age.go
package space

import "strings"

type Planet string

var earthYear = map[Planet]float64{
	"mercury": 0.2408467,
	"venus":   0.61519726,
	"earth":   1.0, // 365.25 Earth days, or 31557600 seconds
	"mars":    1.8808158,
	"jupiter": 11.862615,
	"saturn":  29.447498,
	"uranus":  84.016846,
	"neptune": 164.79132,
}

func secondsToYears(seconds float64) float64 {
	return seconds / (60 * 60 * 24 * 365.245)
}

func Age(seconds float64, planet Planet) float64 {
	p := strings.ToLower(string(planet))
	factor, ok := earthYear[Planet(p)]
	if !ok {
		return -1
	}

	return secondsToYears(seconds) / factor
}
